File: rock-paper-scissors/rock_ppr_scissors.py
def win(a,b):
   print(f"Computer choose {change[a]}")
   print("Congratulations you beat the computer!!!")

def loss(a,b):
   print(f"Computer choose {change[a]}")
   print("You loose!!")


import random

x=random.randint(0,2)

change = {
    0 : "Rock" ,
    1 : "Paper" ,
    2 : "Scissors"
}

print("Welcome to the game!!!")

while(1):
   
 y=int(input("Choose 0 for Rock , 1 for paper & 2 for scissors :"))

 if(y>2 or y<0):
    print("Invalid!! Try again!!")

    if(y==0 or y==1 or y==2):
       break
 elif(x==y):
    print("Match tied.. Try again!!")

    if(x!=y):
       break

 else:
      break 

if((x==0 and y==1)  or (x==1 and y==2) or (x==2 and y==0) ):
    win(x,y)

else:
   loss(x,y)

print("Thank you for playing!!!")


   

File: rock-paper-scissors/rock_ppr_scissors.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main()
{
    int player, computer;
    srand(time(0));
    computer = (rand() % 3); // rand() % (range_size) + minimum_value

    /*0----->Rock
      1------> Paper
      2------> Scissor*/
    printf("Welcome to rock paper scissors vs computer! \n");
    while (1)
    {
        printf("Choose 0 for Rock , 1 for Paper and 2 For Scissors :");
        scanf("%d", &player);

        if (player != 0 && player != 1 && player != 2)
        {
            printf("Invalid!! , Only 0,1,2 are allowed\n");
            continue;
        }
        else if (player == computer)
        {
            printf("Game is tied!!\n");
            continue;
        }

        else
        {
            break;
        }
    }
    printf("Computer choose %d \n", computer);

    if ((player == 0 && computer == 2) || (player == 1 && computer == 0) || (player == 2 && computer == 1))
    {
        printf("You beat the computer!!!!!\n");
    }

    else
    {
        printf("Computer wins!!\n");
    }

    printf("Thankyou for playing..");

    return 0;
}